### ISSUE TITLE
Immutable with Jackson annotation @JsonIgnoreProperties with ignore properties failing during deserialization not working  

### ISSUE THREAD

**Issue by shyam4u** [immutables/immutables#603]
If I have an immutable which has a derived field having no setter and I want to ignore the same field during de- serialization am unable to do so .I keep getting unrecognized property exception from jackson .
org.codehaus.jackson.map.exc.UnrecognizedPropertyException:

Below is the sample immutable and my used case 

@JsonIgnoreProperties(value={ "computed" }, allowGetters=true)
@Value.Immutable
@Value.Style(typeImmutable = "MyClassImpl", jdkOnly = true)
@JsonTypeName("myClass")
@JsonDeserialize(builder = MyClassImpl.Builder.class)
public abstract class  MyClass {

	@Value.Derived
	@JsonProperty("type")
	public String getType() {
		return "CATEGORY";
	}
	
	@JsonProperty("name")
	public String getName()
}

I guess we should have the derived property ignored by default while deserialization .We need the serialization to happen as it is a read only property.Is there any way we can achieve this feature using immutable ?

**Comment by elucash** [comment]
Hi, thank you for reporting this. I would gladly try to reproduce the issue, but it seems that you have very old version of Jackson running (1.x) judging from the `org.codehaus.jackson.map.exc.UnrecognizedPropertyException` fully qualified name. Immutables only works with Jackson version 2.5 or higher. Please, untangle the dependencies and let us know if the problem persists.

**Comment by shyam4u** [comment]
No @elucash  I have not pasted the correct stack trace .My version of jackon is 2.5+ and immutables is 2.4+

**Comment by augustotravillio** [comment]
@shyam4u Have you tried `@JsonIgnoreProperties(ignoreUnknown = true)`

**Comment by ldriscoll** [comment]
@augustotravillio @elucash I've also run into something like this.
Basically, `@Derived` causes us to not have any setters for that value in the `Json` class.  The case I've seen is where the getter is for a List; so I think Jackson is calling the getter, to get the list, to add new values to it; and we get the `UnsupportedOperationException`.

`@JsonIgnoreProperties` is not helping in this case.

I've tried annotating the method as
```java
@Value.Derived
@JsonProperty(access = Access.READ_ONLY)
public List<String> getThings() {
...
```
and this also doesn't work. 

Basically, we need a way to have `@Value.Derived` that can be serialised, but not fail during deserialisation (since we have javascript front ends that need the data there)

**Comment by jaymmodi** [comment]
Still an issue, I had to manually had an annotation on my field `@Nullable` for Immutables to work with JsonIgnoreProperties.
Got this from [Immutables doc](https://immutables.github.io/json.html) - 

```
If using @JsonIgnore, you should explicitly make an attribute non-mandatory. In Immutables, an attribute can be declared as non-mandatory via @Nullable, Optional or @Value.Default which are all different in their effect and we do not derive anything automatically.
```
